Extension/Services/TroopActionService.cs

```diff
@@ -78,25 +78,39 @@ public static (int, int) RecruitPrisoners(PartyVM partyVm, PartyScreenLogic part
                 }
             }
 
+            var playerSelectedCharacter = partyVm.CurrentCharacter.Character;
+
             int recruitedTypes = 0;
             int recruitedTotal = 0;
             List<Tuple<PartyCharacterVM, PartyScreenLogic.PartyCommand>> commands = new List<Tuple<PartyCharacterVM, PartyScreenLogic.PartyCommand>>();
-            while (recruitablePrisoners.Count > 0 && (partySpace > 0 || States.HotkeyControl))
+            foreach (var prisoners in recruitablePrisoners.OrderByDescending(vm => vm.Character.Tier))
```

Comment:
I don't think this is needed, as the recruiting should respect the sorting the player has setup. I think this is something we can expand on later as a setting. e.g Recruiting respects sort, recruits highest tier first, etc similar to how you have your upgradable rules

Reply:
Forgot the sorting was a big part of this mod ;) removing orderby